elasticsearch exception when persisting dc.date

When the document plugin is enabled it attempts to find Dublin Core metadata to persist as part of the annotation. When it finds a dc.date it persists it as annotation.document.dc.date which ElasticSearch tries to helpfully parse as a date. This is nice except when the date isn't in a format that ElasticSearch recognizes, and it throws an exception, which causes the annotation save to fail. 

For example when annotating [this page](http://adsabs.harvard.edu/abs/2006Sci...311.1393P) which has a dc.date of "2006-03" which throws this exception in the ElasticSearch:

```
Caused by: org.elasticsearch.index.mapper.MapperParsingException: failed to parse date field [03/2006], tried both date format [dateOptionalTime], and timestamp number
    at org.elasticsearch.index.mapper.core.DateFieldMapper.parseStringValue(DateFieldMapper.java:426)
    at org.elasticsearch.index.mapper.core.DateFieldMapper.innerParseCreateField(DateFieldMapper.java:378)
    at org.elasticsearch.index.mapper.core.NumberFieldMapper.parseCreateField(NumberFieldMapper.java:167)
    at org.elasticsearch.index.mapper.core.AbstractFieldMapper.parse(AbstractFieldMapper.java:385)
    ... 14 more
Caused by: java.lang.IllegalArgumentException: Invalid format: "03/2006" is malformed at "/2006"
```

--- a/annotator/annotation.py
+++ b/annotator/annotation.py
@@ -41,6 +41,15 @@ MAPPING = {
                     'type': {'type': 'string', 'index': 'not_analyzed'},
                     'href': {'type': 'string', 'index': 'not_analyzed'},
                 }
+            },
+            'dc': {
+                'type': 'nested',
+                'properties': {
+                    # by default elastic search will try to parse this as 
+                    # a date but unfortunately the data that is in the wild 
+                    # may not be parsable by ES which throws an exception
+                    'date': {'type': 'string', 'index': 'not_analyzed'}
+                }
             }
         }
     }

--- a/annotator/document.py
+++ b/annotator/document.py
@@ -5,6 +5,7 @@ MAPPING = {
     'annotator_schema_version': {'type': 'string'},
     'created': {'type': 'date'},
     'updated': {'type': 'date'},
+    'title': {'type': 'string'},
     'link': {
         'type': 'nested',
         'properties': {
@@ -12,7 +13,15 @@ MAPPING = {
             'href': {'type': 'string', 'index': 'not_analyzed'},
         }
     },
-    'title': {'type': 'string'}
+    'dc': {
+        'type': 'nested',
+        'properties': {
+            # by default elastic search will try to parse this as 
+            # a date but unfortunately the data that is in the wild 
+            # may not be parsable by ES which throws an exception
+            'date': {'type': 'string', 'index': 'not_analyzed'}
+        }
+    }
 }
 
 class Document(es.Model):

